contrib/node/src/python/pants/contrib/node/tasks/node_bundle.py

```diff
@@ -0,0 +1,61 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.base.build_environment import get_buildroot
+from pants.fs import archive
+
+from pants.contrib.node.tasks.node_paths import NodePaths
+from pants.contrib.node.tasks.node_task import NodeTask
+
+
+class NodeBundle(NodeTask):
+  """Create an archive bundle of NodeModule targets."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(NodeBundle, cls).register_options(register)
+    # Some node modules depend on symlink to function.  Thus we can only support archival type
+    # that can preserve symlinks.
+    register('--archive', choices=list(archive.TYPE_NAMES_PRESERVE_SYMLINKS),
```

Comment:
It seems unlikely that every `node_module` that is defined would also require a `node_bundle` target, so I'm not sure how high the overhead would be in practice.

Does `npm` differentiate between binary and library `package.json`s? If not, then I think that adding a `node_bundle` target would make sense, as it would allow you to explicitly indicate which target in your build graph is a root.

But if there is uncertainty here, would bias toward not adding the target argument for now.